### PR TITLE
6965 odata parser any() and all()

### PR DESCRIPTION
**Pull request by wiedld**
See detailed what/why in issue: 
https://github.com/lanetix/issues/issues/6965

### CODE REVIEW

**Comment by tone81** [comment]
@wiedld Was file formatting the cause of the 10000+ diff in `lib/odata-parser.js`? Or is there code changes to note in the file?

**Comment by wiedld** [comment]
@tone81 
lib/odata-parser.js is an auto-generated file using pegjs.
After updating the BNF, I autogenerated and commited the new file.

That's why I really really need to write a test case for this one before we approve.
But I wanted to share this, before getting your CR with the odata-mapper FIXME (still in progress).

**Comment by wiedld** [comment]
Looking at auto-gen, it looks really different. I may have a different version of pegjs. Let me fix that.

Thanks for catching.

**Comment by wiedld** [comment]
auto-gen file now:
```+121 −17```

**Comment by pythonesque** [comment]
👍 !

File: test/parser.specs.js [hunk 1]
```diff
@@ -232,6 +232,43 @@ describe('odata.parser grammar', function () {
 
     });
 
+    it('should parse any(lambdaFunc) eq true $filter', function () {
+
+        var ast = parser.parse("$filter=linked_table/any_num_hops/string_list/any(list_item:list_item eq 'test') eq true");
+
+        // assert.equal(ast.$filter.type, "eq");
```
Comment (tone81) [inline test/parser.specs.js]: Remove.

File: test/parser.specs.js [hunk 2]
```diff
@@ -232,6 +232,43 @@ describe('odata.parser grammar', function () {
...
+
+        assert.equal(ast.$filter.right.value.true);
+
+    });
+
+    it('should parse all(lambdaFunc) eq true $filter', function () {
+
+        var ast = parser.parse("$filter=linked_table/any_num_hops/string_list/all(list_item:list_item eq 'test') eq true");
+
+        // assert.equal(ast.$filter.type, "eq");
```
Comment (tone81) [inline test/parser.specs.js]: Remove.

File: src/odata.pegjs [hunk 3]
```diff
@@ -367,6 +367,55 @@ otherFunc1                  = f:otherFunctions1Arg "(" arg0:part ")" {
...
+                                      path: p.idPath,
+                                      type: "functioncall",
+                                      func: p.func,
+                                      args: [arg0]
+                                  }
+                              }
+
+// pegjs has a problem handling identifierPath/any(). This gets around it.
+idPathANDfuncArgExpr       = a:identifier b:idPartANDfuncArg* {
+                                 // thorw if the last item in not a collectionFunction name
```
Comment (tone81) [inline src/odata.pegjs]: thorw 😂

**Comment by tone81** [comment]
Nothing egregious in the PR. 🚀

**Pull request merged**